### PR TITLE
Make README contain more useful information about the project

### PR DESCRIPTION
**Pull request by cahirwpz**
Make the main page readme file contain high-level overview of the project, in order to encourage the reader to learn more about Mimiker project.

### CODE REVIEW

File: wiki/docs/toolchain.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Toolchain
+---
+
+To build Mimiker you will need a custom toolchain we use. It contains cross
+compiler (`gcc`), linker (`binutils`) and debugger (`gdb`). We prepared prebuild
```
Comment (pj1031999) [inline wiki/docs/toolchain.md:5]: Is it still true after transition to llvm?

File: wiki/docs/running.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+---
+
+We provide a Python script, i.e. `launch`, that simplifies running Mimiker OS.
+The kernel image is run with QEMU or Renode simulator. One or more serial
+consoles are available for interaction. Optionally you can attach to simulator
+with `gdb` debugger.  All of that is achieved by running all interactive
+sessions within [tmux](https://github.com/tmux/tmux/wiki) terminal multiplexer
+with default key bindings.
+
+In project main directory, run command below that will start the kernel in
```
Comment (pj1031999) [inline wiki/docs/running.md:11]: s/main/root/ imho (you already use root in building.md)

File: README.md [hunk 3]
```diff
@@ -1,108 +1,108 @@
...
+and interrupts in similar manner to FreeBSD's [NewBus][14]. Special care is
+taken to make drivers portable. We have enumerator routines that autodetect
+devices attached to PCI and USB buses. We use [flat device tree][13] to drive
+kernel configuration during startup phase.
+
+Virtual file system and user virtual address space management are loosely based
+on FreeBSD ideas. They need substatial amount of work to become as mature as in
+FreeBSD or Linux kernels.
+
+## What we're proud of ?
```
Comment (pj1031999) [inline README.md:64]: What we're proud of ? -> What we're proud of?
I'm not sure which looks better

**Comment by cahirwpz** [comment]
@pj1031999 @MichalBlk Please edit the files as you wish.

File: wiki/docs/toolchain.md (hunk 1 above)
Comment (cahirwpz) [reply wiki/docs/toolchain.md:5]: It's still valid. For some period of time GNU toolchain will be second class citizen before support for it is removed.

File: README.md (hunk 3 above)
Comment (korbiniak) [reply README.md:64]: IMHO space before '?' looks bad. Again, shouldn't it be "What are we proud of?", or simply "What we are proud of" (without question mark).

File: README.md [hunk 4]
```diff
@@ -1,108 +1,108 @@
...
+
+Though userspace programs are part of Mimiker project, they've got simply ported
+from NetBSD or [suckless][1] project. We focus on kernel development, since we
+find it more interesting. We don't want to invest too much time into the device
+drivers, so we keep a list of target platforms small.
+
+If you'd like to get involved in the project please read our [Wiki][2] to find
+out more!
+
+## Where we are?
```
Comment (korbiniak) [inline README.md:27]: Shouldn't it be "Where are we?" (or imho better: "Where we are", without the question mark).

File: README.md [hunk 5]
```diff
@@ -1,108 +1,108 @@
...
+
+Mimiker supports:
+ * UNIX file I/O -- well known APIs for file-like objects access,
+ * interprocess communication -- POSIX signal and pipes,
+ * job control -- thus we can run unmodified [Korn Shell][18],
+ * UNIX credentials -- users, groups, file permissions,
+ * libterminfo, hence Mimiker can run some fullscreen terminal applications,
+ * [pseudoterminals][16] -- so we can run [script][17] or terminal emulators.
+
+## What is missing?
```
Comment (korbiniak) [inline README.md:80]: ditto, imho "What is missing" looks nicer

File: wiki/docs/Home.md [hunk 6]
```diff
@@ -22,4 +22,14 @@ When you design and implement an interface, please keep in mind following practi
...
+
+### Documentation
+
+Useful sites:
+* [OSDev wiki](http://wiki.osdev.org)
+
+Toolchain documentation:
+* [Extensions to the C Language Family](https://gcc.gnu.org/onlinedocs/gcc/C-Extensions.html)
+* [Debugging with GDB](https://sourceware.org/gdb/onlinedocs/gdb/index.html)
+* [Linker scripts](https://sourceware.org/binutils/docs/ld/Scripts.html)
```
Comment (korbiniak) [inline wiki/docs/Home.md:35]: General comment to this file: is the first sentence on the page necessary? I remember @cahirwpz telling me it's unimportant, so why keep it there? Also, I imagine that most people that come to read wiki do it out of curiosity, so starting the home page with "Before you pick up any task [...]" seems strange to me. Shouldn't it begin with some very brief overview of what Mimiker is and what can be found in the wiki?

File: wiki/docs/building.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
+Building
+---
+
+With toolchain installed you are ready to compile Mimiker. Before you run
```
Comment (korbiniak) [inline wiki/docs/building.md:4]: Maybe `toolchain` -> `[toolchain](wiki/docs/toolchain.md)`

File: wiki/docs/running.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
+Running
+---
+
+We provide a Python script, i.e. `launch`, that simplifies running Mimiker OS.
```
Comment (korbiniak) [inline wiki/docs/running.md:4]: Maybe add a link to the script?

File: wiki/docs/building.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+`make` in project's root directory, please have a look at `config.mk` file.
+
+Of several configurable options, the following are most important to you:
```
Comment (korbiniak) [inline wiki/docs/building.md:7]: "the following are most important to you" -- this seems strange to me (how can you know what is important to me?), maybe "the following are the most interesting"? Or maybe "There are several configurable options, including:"

File: wiki/docs/building.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
+Building
+---
+
+With toolchain installed you are ready to compile Mimiker. Before you run
+`make` in project's root directory, please have a look at `config.mk` file.
```
Comment (korbiniak) [inline wiki/docs/building.md:5]: Why should I have a look there? What is important in `config.mk`? Maybe elaborate on that.

File: wiki/docs/running.md (hunk 2 above)
Comment (korbiniak) [reply wiki/docs/running.md:11]: I'd rephrase this sentence: "To start kernel in test-run mode, run the following command in the project's root directory:
<command here>
To finish simulation [...]

**Comment by pj1031999** [comment]
> @pj1031999 @MichalBlk Please edit the files as you wish.

It can work if you have comments only from one person. If you have more comments and some of them are in conflict... you must be a person who decide which way is better ;)

File: wiki/docs/toolchain.md (hunk 1 above)
Comment (pj1031999) [reply wiki/docs/toolchain.md:5]: Huh? When I'm reading this paragraph I have the following thoughts:
  * `gcc`, `ld.bfd` and `gdb` are *required* to build & run Mimiker.
  * I can run it only on x86-64 not specified version of Debian using `qemu` with unknown patches.

First is not true. You can run Mimiker without `gdb` - probably it's not what you want to achieve... but you can.
`gcc` is *not* required - since we support `llvm` - personally I don't have custom gnu toolchain and I'm using only default clang which is enough to build.
We should mark that all patches to qemu (and other toys that we are using) are open-source and they are located in toolchain directory and end user can build everything using our build system.

File: wiki/docs/Home.md (hunk 6 above)
Comment (cahirwpz) [reply wiki/docs/Home.md:35]: Let's target this issue later. I don't want to grow this PR indefinitely.

**Comment by cahirwpz** [comment]
@korbiniak @pj1031999 Please re-review.

File: README.md [hunk 11]
```diff
@@ -1,108 +1,108 @@
...
+find it more interesting. We don't want to invest too much time into the device
+drivers, so we keep a list of target platforms small.
+
+If you'd like to get involved in the project please read our [Wiki][2] to find
+out more!
+
+## Where we are
+
+Mimiker is a real-time operating system. The kernel is preemptible and our
+mutexes support priority inheritance. We minimize work done in interupt context
```
Comment (hadarai) [inline README.md:30]:
```suggestion
mutexes support priority inheritance. We minimize work done in interrupt context
```

**Review by pj1031999 — APPROVED** [review]
